lib/matplotlib/tests/test_collections.py

```diff
@@ -899,17 +899,23 @@ def test_collection_set_array():
 
 
 def test_blended_collection_autolim():
-    a = [1, 2, 4]
-    height = .2
+    f, ax = plt.subplots()
 
-    xy_pairs = np.column_stack([np.repeat(a, 2), np.tile([0, height], len(a))])
-    line_segs = xy_pairs.reshape([len(a), 2, 2])
+    # sample data to give initial data limits
+    ax.plot([2, 3, 4], [0.4, 0.6, 0.5])
+    np.testing.assert_allclose((ax.dataLim.xmin, ax.dataLim.xmax), (2, 4))
+    data_ymin, data_ymax = ax.dataLim.ymin, ax.dataLim.ymax
 
-    f, ax = plt.subplots()
+    # LineCollection with vertical lines spanning the Axes vertical, using transAxes
+    x = [1, 2, 3, 4, 5]
+    vertical_lines = [np.array([[xi, 0], [xi, 1]]) for xi in x]
     trans = mtransforms.blended_transform_factory(ax.transData, ax.transAxes)
-    ax.add_collection(LineCollection(line_segs, transform=trans))
-    ax.autoscale_view(scalex=True, scaley=False)
-    np.testing.assert_allclose(ax.get_xlim(), [1., 4.])
+    ax.add_collection(LineCollection(vertical_lines, transform=trans))
+
+    # check that the x data limits are updated to include the LineCollection
+    np.testing.assert_allclose((ax.dataLim.xmin, ax.dataLim.xmax), (1, 5))
+    # check that the y data limits are not updated (because they are not transData)
+    np.testing.assert_allclose((ax.dataLim.ymin, ax.dataLim.ymax), (0.4, 0.6))
```

Comment:
Note 1: Before, the test did only check that the x data limits were expanded. It did not investigate the y data limits. Now it also ensures, that y data limits are not changed.

Note 2: I've switched from testing view limits to testing data limits, because that's the relevant quantity autolim influences. We save the `autoscale_view()` by not looking at the view limits.